packages/bee-q/src/components/dropdown/bq-dropdown.tsx

```diff
@@ -0,0 +1,164 @@
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentDidLoad() {
+    this.initFloatingUIFromPanel();
+  }
+
+  // Listeners
+  // ==============
+
+  /** On click outside the panel */
+  @Listen('click', { target: 'document', passive: true })
+  onClickOutsidePanel(event: MouseEvent) {
+    // close the panel on click, except click within the panel or within the trigger element
+    if (!event.composedPath().includes(this.panelElement) && !event.composedPath().includes(this.trigger)) {
+      this.panelElement.open = false;
+    }
+  }
+
+  /**
+   * If the user press on Escape button
+   * or `<bq-dropdown>` loses focus
+   * then hide the panel
+   */
+  @Listen('keyup', { target: 'window', passive: true })
+  onEscape(event: KeyboardEvent) {
+    if (event.key === 'Escape' || (event.key === 'Tab' && !event.composedPath().includes(this.el))) {
+      this.panelElement.open = false;
+    }
+  }
+
+  @Listen('bqPanelVisibility', { passive: true })
+  onPanelVisibilityChange(event: CustomEvent) {
+    const isOpened: boolean = event.detail as boolean;
+
+    this.bqPanelChange.emit({ opened: isOpened });
+  }
+
+  @Listen('bqSelect', { passive: true })
+  onItemSelect() {
+    this.panelElement.open = false;
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private openPanel = (): void => {
+    this.panelElement?.togglePanel();
+    this.panelElement.open = !this.panelElement.open;
+  };
+
+  private initFloatingUIFromPanel = (): void => {
+    this.panelElement?.setTriggerElement(this.el);
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <div class="bq-dropdown" part="base">
+        {/* TRIGGER ELEMENT */}
+        <div
+          class="bq-dropdown__trigger"
+          ref={(el) => (this.trigger = el)}
+          onClick={this.openPanel}
+          aria-haspopup="true"
+          aria-expanded={this.panelOpen}
+          part="trigger"
+        >
+          <slot name="trigger" />
+        </div>
+        {/* PANEL */}
+        <bq-panel
+          class="bq-dropdown__panel"
+          distance={this.panelDistance}
+          placement={this.panelPlacement}
+          open={this.panelOpen}
+          scrollbar={this.panelScrollbar}
+          ref={(el) => (this.panelElement = el)}
+          aria-labelledby="dropdown"
+          role="region"
+          part="panel"
+        >
+          <slot />
+        </bq-panel>
+      </div>
```

Comment:
Some Tailwind CSS utilities can be added directly in the JSX, no need to use `.scss` for a couple of rules.

```suggestion
      <div class="bq-dropdown" part="base">
        {/* TRIGGER ELEMENT */}
        <div
          class="bq-dropdown__trigger block"
          ref={(el) => (this.trigger = el)}
          onClick={this.togglePanel}
          aria-expanded={this.open ? 'true' : 'false'}
          aria-haspopup="true"
          part="trigger"
        >
          <slot name="trigger" />
        </div>
        {/* PANEL */}
        <bq-panel
          class="bq-dropdown__panel"
          distance={this.distance}
          placement={this.placement}
          open={this.open}
          sameWidth={this.sameWidth}
          skidding={this.skidding}
          strategy={this.strategy}
          triggerElement={this.trigger}
          aria-labelledby="dropdown"
          role="region"
          part="dropdown"
          exportparts="panel"
        >
          <slot />
        </bq-panel>
      </div>
```